app/frontend/javascript/custom_google_analytics_4_events.js

```diff
@@ -8,21 +8,44 @@ $(document).on('click', '*[data-analytics-category]', function(e) {
   // Do not call `gtag` unless the function is defined
   // in app/views/layouts/_google_analytics_4.html.erb .
   // (This in turn is controlled by ScihistDigicoll::Env.lookup(:google_analytics_4_tag_id).
+
+
   if (typeof gtag === 'function') {
 
-    gtag( 'event',
+    // In certain cases we want to set event_value to the contents of a text input - one that
+    // might be different every time the link or button is clicked.
+
+    // Example use case: we want to set event_value to the query string in search-within-the-work search form.
+
+    // If we find a selector in data-analytics-value-css, we can look up the current value of the text input 
+    // using that selector, and then pass that string to GA as event_value.
+    var elementToLookUp =  e.currentTarget.getAttribute("data-analytics-value-css");
+    var eventValue = elementToLookUp ?  $(elementToLookUp)[0].value.replace(/[^a-zA-Z 0-9]+/g, '') : null;
+
+    var data_to_send = {
+      // Always the string "work".
+      'event_category': e.currentTarget.getAttribute("data-analytics-category"),
+
+      // Always the work's friendlier_id.
+      'event_label': e.currentTarget.getAttribute("data-analytics-label"),
 
-      // param eventName
+      // Only used to send a search phrase (or, more frequently, null).
+      'event_value': eventValue
+    }
+
+    // Based on https://support.google.com/analytics/answer/13675006?hl=en, let's try a more user-friendly format as well:
+    if (eventValue) {
+      data_to_send['search_phrase'] = eventValue;
+      data_to_send['friendlier_id'] = e.currentTarget.getAttribute("data-analytics-label");
+    }
```

Comment:
OK, I know I was sort of suggesting this, but i was suggesting changing the `data-` attribute to something specific like "search phrase" too. 

It actually doesn't work to just assume that `data-analytics-value` is ALWAYS a search phrase, like undocumented API that when we say "send a value" we *really* mean "send a search phrase", and will record it in our analytics as such. 

And same for assuming that `data-analytics-label` *always* means "friendlier-id".  If that were true, we should just make the html data attribute (which is up to us), `data-analytics-friendlier-id`, right?

Are you following me? We need to think of those `data-` attributes as the "API" to our "record in GA" routine, and the API needs to be transparent and legible.  

We can talk about this more next week?